source/extensions/access_loggers/file/config.cc

```diff
@@ -29,30 +30,33 @@ FileAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
       config, context.messageValidationVisitor());
   AccessLog::FormatterPtr formatter;
 
-  if (fal_config.access_log_format_case() == envoy::extensions::access_loggers::file::v3::
-                                                 FileAccessLog::AccessLogFormatCase::kFormat ||
-      fal_config.access_log_format_case() ==
-          envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
-              ACCESS_LOG_FORMAT_NOT_SET) {
-    if (fal_config.format().empty()) {
-      formatter = AccessLog::AccessLogFormatUtils::defaultAccessLogFormatter();
-    } else {
-      formatter = std::make_unique<AccessLog::FormatterImpl>(fal_config.format());
+  if (fal_config.has_log_format()) {
+    formatter = SubstitutionFormatStringUtils::fromProtoConfig(fal_config.log_format());
+  }
+  // backward compatible code for deprecated access_log_format, to be removed.
+  if (fal_config.access_log_format_case() !=
+      envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
+          ACCESS_LOG_FORMAT_NOT_SET) {
+    envoy::config::core::v3::SubstitutionFormatString sff_config;
+    switch (fal_config.access_log_format_case()) {
+    case envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::kFormat:
+      sff_config.set_text_format(fal_config.format());
+      break;
+    case envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
+        kJsonFormat:
+      *sff_config.mutable_json_format() = fal_config.json_format();
```

Comment:
Not sure if this is a breaking change.  current "json_format" users are getting JSON format values as string type.  E.g. 
{ "code":  "200" }.   With this change, we retain the proper types for bool and int,  so it becomes {"code": 200}. 

This field is only used in file access_log. It will not break envoy,  if it ever break, it will only break file access log JSON parser.

Reply:
Technically I think this is a breaking change, though we have had bugs filed in the past with people complaining about this behavior. Because of that it would not surprise me if people are currently expecting this behavior and this would break them. Is there any way to not change this as part of this refactor? cc @junr03 @dio who have dealt with similar issues.

Reply:
In that case, I will keep "typed_json_format" and support both "json_format" and "typed_json_format" as it is now.